dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -113,42 +144,65 @@ export default defineComponent({
      */
     updateSelectedReports(reports: ReportToUpload[]) {
       this.reportsToUpload = reports;
-      if (this.duplicatesAmongReferenceableReportNames()) {
-        const indexOfLastSelectedFile = reports.length - 1;
-        const lastSelectedFile = reports[indexOfLastSelectedFile].file;
-        this.openModalToDisplayDuplicateNameError(lastSelectedFile.name);
-        (this.$refs.uploadDocumentsForm.removeDocumentFromDocumentsToUpload as (index: number) => void)(
-          indexOfLastSelectedFile,
-        );
+
+      if (this.areDuplicatesAmongReferenceableReportNames()) {
+        const foundExistingRecords = new Set<string>();
+        const duplicatesWithIndex = [];
+
+        for (let i = 0; i < this.reportsToUpload.length; i++) {
+          const currentName = this.reportsToUpload[i].fileNameWithoutSuffix;
+
+          if (foundExistingRecords.has(currentName) || this.namesOfStoredReports.indexOf(currentName) !== -1) {
+            duplicatesWithIndex.push({ report: this.reportsToUpload[i], index: i });
+          } else {
+            foundExistingRecords.add(currentName);
+          }
+        }
+
+        this.handleReportDuplicates([...duplicatesWithIndex].reverse());
       } else {
-        this.emitReferenceableReportNamesChangedEvent();
+        this.emitReportsUpdatedEvent();
       }
     },
+    /**
+     * Scan list of file names and show modal if duplicate
+     * @param duplicatesWithIndex a list of reports to upload
+     */
+    handleReportDuplicates(duplicatesWithIndex: { report: ReportToUpload; index: number }[]) {
+      duplicatesWithIndex.forEach(({ report, index }) => {
+        this.openModalToDisplayDuplicateNameError(report.fileNameWithoutSuffix);
+        (this.$refs.uploadDocumentsForm.removeDocumentFromDocumentsToUpload as (index: number) => void)(index);
+      });
```

Comment:
This has a bug:

If I select 3 reports with names that are already uploaded:

![image](https://github.com/d-fine/Dataland/assets/99255296/911a46c9-3571-4bdb-bc98-1acd0fa09676)

I get 6 modals that overlap:

![image](https://github.com/d-fine/Dataland/assets/99255296/9927caff-f0a4-4da1-ba41-974bc640cc5c)

The overlapping isn't that bad, but it should be 3 modals, right?  Not 6. 

And the crazy thing is: I get 3 modals for Sustainibility Report, 2 modals for IntegratedReport, and 1 for ESEFReport .

Optimal solution would be to name all 3 duplicate reports in one modal.
If not possible, than at least one modal per duplicate.

Reply:
There should be one modal with a list of duplicate files. I'll have a look.

Reply:
Still working on a full fix... currently managed to combine duplicates into one modal. However, the modal is still opened a number of times.. per file. Still working on this...

Reply:
Done. Please test.

Reply:
I checked in other frameworks, too. Seems fine. But curious what you think.

Reply:
... actually... There may be an index deletion bug. I need to check. Please hold.

Reply:
This change currently leads to an sonar code smell i fear

Reply:
Ok I will test this now.   I think this bug should get a test.

Reply:
Works fine. Thx.  Just a small request. Could you change "file" to "file(s)" in the text?:

![image](https://github.com/d-fine/Dataland/assets/99255296/8775b773-61d4-4821-b7e9-194d97e91fa2)

Reply:
> Ok I will test this now. I think this bug should get a test.

What exactly would you want to get tested here? That there is only one modal opening or that there are multiple reports mentioned in the modal?

Reply:
Well obviously this thing was broken for more than one duplicate file, and I noticed it with much luck. Therefore a test is not a bad idea.
Therefore I suggest one of these two options:

a) Extend the existing e2e-test, which currently only uploads one duplicate (and therefore did not detect this error), to upload two duplicates. => Check if modal appears with both duplicate names in it. And if you close it, there is no modal anymore.

or 

b) Write a new component test which tests this.

Reply:
I'd go with option a) .. both due to simplicity and time constraints.
I'll get right on it.

Reply:
Ok that's fine for me.

As I mentioned, there is already an e2e test in place which tests the duplicate modal.  But it does it only for one file. 
So just extend/adjust it to check it for two files, and that should be enough.

Reply:
> Well obviously this thing was broken for more than one duplicate file, and I noticed it with much luck. Therefore a test is not a bad idea. Therefore I suggest one of these two options:
> 
> a) Extend the existing e2e-test, which currently only uploads one duplicate (and therefore did not detect this error), to upload two duplicates. => Check if modal appears with both duplicate names in it. And if you close it, there is no modal anymore.
> 
> or
> 
> b) Write a new component test which tests this.

Ok, just wanted to make sure, that we do not want to test some weird behavior like "check that only one modal appears" or something in this direction.

Reply:
Well we implicitly check that already, because we "X" the modal that appears in the test.
If there were several modals, then after "X"-ing the first one, you would not see the site again.

Reply:
Yeaht thats true.

Reply:
Done. Added to `\tests\e2e\specs\eu-taxonomy-financials\FileUploadAndLinking.ts` line 73

Reply:
Like it, thx !